docs: SDK architecture + small changes (#435)
GetStream/stream-video-js: packages/react-sdk/docusaurus/docs/React/01-basics/07-ui-customization-options.mdx

## Introduction

There are different ways to use the React Video SDK to build out UI. It comes with powerful built-in components ready for use.
In addition, it allows for styling and custom theming options with custom CSS.

> For custom use cases it is also possible to create your own components to fully tailor the SDK towards your needs. Please note that we suggest not to override a few, core components as those contain complex, low-level logic related to performance, the list of these components can be found in the [UI Components overview](../03-ui/01-overview.mdx).

Reviewer: ```suggestion
For custom use cases it is also possible to create your own components to fully tailor the SDK to your needs. Please note that we suggest not to override a few, core components as those contain complex, low-level logic related to performance, the list of these components can be found in the [UI Components overview](../03-ui/01-overview.mdx).
```

This page guides you through all of them with increasing levels of customization.

## Using pre-built components

> The SDK comes with a lot of pre-built components for the different elements of the UI.

Reviewer 2: But their customizability will be limited as we do not plan to allow passing Components to props.

Reviewer 2: I am not sure how they will be able to customize the pre-built components.

Reviewer: Some of them offer some props to tweak them. In any case there are options for styling them via CSS properties, right?

Reviewer 2: I would conside the CSS part on the following section. I am still not sure how customizable they will be. Our intention is not to follow the stream-chat-react pattern, where you pass down components in props.

Reviewer 2: Ok, let's revisit this later

:::tip
For a list of all available UI components head over to the [Components](../../ui/01-ui-overview) section.
:::

## Theming with CSS

If you only want to customize the styling of the available components, there is a wide range of CSS variables that you can override.

> With that, you can achieve a significant amount of theming in your video product.

Reviewer 2: The following section has a grammatic mistake:

Here are a few examples: -> Here are few examples:

Author: I think this page, in general, will need some more work, as mentioned by @DaemonLoki. Feel free to add those changes @MartinCupela, or we can revisit them outside of this PR.

Reviewer: Agree with Zita. 

It surely requires info (e.g. I didn't know about the `str-video` thing up until yesterday. So there's surely more things needed, I just wanted to give a basic skeleton to have a start.

Please always feel free to fill this with relevant wisdom! :)

Reviewer: > The following section has a grammatic mistake:
> 
> Here are a few examples: -> Here are few examples:

I'd argue that the first one is actually correct?

Reviewer 2: few vs. a few (https://www.grammarly.com/blog/few-a-few/) - not sure which you had in mind @DaemonLoki

There are different ways to use the React Video SDK to build out UI. It comes with powerful built-in video components ready for use.
In addition, it allows for styling and custom theming options with custom CSS.

For custom use cases it is also possible to create your own components to fully tailor the SDK towards your needs. Please note that we suggest not to override a few, core components as those contain complex, low-level logic related to performance, the list of these components can be found in the [UI Components overview](../03-ui/01-overview.mdx).

> This page guides you through all of them with increasing levels of customization.

Reviewer 2: What the adverb "them" refers to?

Author: hm, I can't seem to find "them" in the text above, if you mean the "the list of these" that refers to the core components referenced in the previous part of the sentence, it seemed to clear to me when I was writing it

Reviewer 2: This page guides you through all of **them** with increasing levels of customization.

Author: Ah, I looked only to my changes, that line wasn't written by me, but from the context them refers to the different customization levels.

Reviewer: With more info in the paragraph before we should maybe change this for more clarity:

```suggestion
This page guides you through all the options you have with increasing levels of customization.
```

## Using pre-built components

The SDK comes with a lot of pre-built components for the different elements of the UI.

:::tip

> For a list of all available UI components head over to the [Components](../../ui/01-ui-overview) section.

Reviewer 2: ```suggestion
For a list of all available UI components head over to the [Components](../../03-ui/01-overview) section.
```

Author: The link works without the changes, I think there are two link formats: one is URL based and the other is file based
